util/newArchStatus.ts

```diff
@@ -6,22 +6,52 @@ export enum NewArchSupportStatus {
   Untested = 'untested',
 }
 
-export function getNewArchSupportStatus({ newArchitecture, github, expoGo }: Library) {
+//new function  to fix new architecture support detection
+export function getNewArchSupportStatus({
+  newArchitecture,
+  github,
+  expoGo,
+  lastPublish,
+  maintained,
+  reactNativeVersion,
+}: Library) {
   // Assume untested unless indicated otherwise through one of the following tests
   let flag = undefined;
 
+  // If the library is not maintained, it likely doesn't support the New Architecture
+  if (maintained === false) {
+    return NewArchSupportStatus.Unsupported;
+  }
```

Comment:
Unfortunately, this is an incorrect assumption. Unmaintained libraries can support New Architecture in multiple cases, like:
* deprecation has happen after New Architecture introduction
* package is JS only (does not include native code)
* package has been marked as unmaintend due to name change